src/common/time/src/interval.rs

```diff
@@ -0,0 +1,708 @@
+        let seconds = (microseconds - (microseconds % 1_000_000)) / 1_000_000;
+        let microseconds = microseconds - seconds * 1_000_000;
+        IntervalFormat {
+            years,
+            months,
+            days,
+            hours,
+            minutes,
+            seconds,
+            microseconds,
+        }
+    }
+}
+
+impl IntervalFormat {
+    /// Convert IntervalFormat to Interval Type
+    pub fn try_into_interval(self) -> crate::error::Result<Interval> {
+        let months = self
+            .years
+            .checked_mul(12)
+            .and_then(|years| self.months.checked_add(years));
+        let microseconds = self
+            .hours
+            .checked_mul(60)
+            .and_then(|minutes| self.minutes.checked_add(minutes))
+            .and_then(|minutes| minutes.checked_mul(60))
+            .and_then(|seconds| self.seconds.checked_add(seconds))
+            .and_then(|seconds| seconds.checked_mul(1_000_000))
+            .and_then(|microseconds| self.microseconds.checked_add(microseconds));
+        // overflow handle
+        let months = match months {
+            Some(mon) => mon,
+            None => {
+                return ParseIntervalSnafu {
+                    raw: "interval months overflow",
+                }
+                .fail()
+            }
+        };
+        let microseconds = match microseconds {
+            Some(micro) => micro,
+            None => {
+                return ParseIntervalSnafu {
+                    raw: "interval microseconds overflow",
+                }
+                .fail()
+            }
+        };
+
+        Ok(Interval {
+            months,
+            days: self.days,
+            micros: microseconds,
+            unit: Default::default(),
+        })
+    }
+
+    /// All the field in the interval is 0
+    pub fn is_zero(&self) -> bool {
+        self.years == 0
+            && self.months == 0
+            && self.days == 0
+            && self.hours == 0
+            && self.minutes == 0
+            && self.seconds == 0
+            && self.microseconds == 0
+    }
+
+    /// Determine if year or month exist
+    pub fn has_year_month(&self) -> bool {
+        self.years != 0 || self.months != 0
+    }
+
+    /// Determine if day exists
+    pub fn has_day(&self) -> bool {
+        self.days != 0
+    }
+
+    /// Determine time part(includes hours, minutes, seconds, microseconds) is positive
+    pub fn has_time_part_positive(&self) -> bool {
+        self.hours > 0 || self.minutes > 0 || self.seconds > 0 || self.microseconds > 0
+    }
+
+    // time part means hours, minutes, seconds, microseconds
+    pub fn has_time_part(&self) -> bool {
+        self.hours != 0 || self.minutes != 0 || self.seconds != 0 || self.microseconds != 0
+    }
+
+    /// Convert IntervalFormat to iso8601 format string
+    pub fn to_iso8601_string(&self) -> String {
+        if self.is_zero() {
+            return "PT0S".to_owned();
```

Comment:
nitpick: Personally, I'd like to use `to_string()` instead of `to_owned()` as it is more clear.
```suggestion
            return "PT0S".to_string();
```

Reply:
You could refer to the codecov coverage page for this file
https://app.codecov.io/gh/GreptimeTeam/greptimedb/pull/1952/blob/src/common/time/src/interval.rs